File: test/fixtures/order-service-typescript/src/checkout.ts
import { subtotal } from './cart.js';
import type { Item, Stock } from './cart.js';
import { canFulfil } from './inventory.js';

export function placeOrder(items: Item[], stock: Stock): number {
  if (!canFulfil(items, stock)) throw new Error('An item is out of stock');
  return subtotal(items);
}

File: test/fixtures/order-service-rust/src/checkout.rs
use std::collections::HashMap;
use crate::cart::{Item, subtotal};
use crate::inventory::can_fulfil;

pub fn place_order(items: &[Item], stock: &HashMap<&str, i32>) -> Result<i32, &'static str> {
    if !can_fulfil(items, stock) {
        return Err("An item is out of stock");
    }
    Ok(subtotal(items))
}
